### DISCUSSION TITLE
`RewriteURIs` feature gate status

### DISCUSSION THREAD

**Discussion by dschniepp** [Kong/kubernetes-ingress-controller#6355] [General]
Hello, how is the process for feature gates to get promoted to GA? Especially I am curious about the future of `RewriteURIs `.

**Comment by programmer04** [comment]
Feature gates are removed during major releases because it's a breaking change. Configuration of not supported (or removed) feature gate stops controller from booting

> If the above answers your question please click Mark as answer button to increase visibility for others.

**Reply by dschniepp** [reply]
@programmer04 thanks, I assumed so, too. But somehow it didn't get removed with version 3.x.x https://docs.konghq.com/kubernetes-ingress-controller/latest/reference/feature-gates/.

**Reply by programmer04** [reply]
Sorry, I wasn't precise. Of course, besides the release of the major version, the team has to decide that the feature is mature enough too in terms of API, stability, etc. So basically, the release of a major version should be regarded as a necessary but not sufficient condition for removing a feature gate
